c++/kmp.cc

```diff
@@ -0,0 +1,78 @@
+/* C++: Knuth-Morris-Pratt (KMP) string matching algorithm
+   =================================================================
+   Description
+
+   Complexity (m = length of pattern, n = length of text)
+   Algorithm  | Pre-processing time  | Matching time
+   -----------+----------------------+------------------------------
+   Naive      | 0                    | O((n+m+1)m)
+   KMP        | O(m)                 | O(n)
+
+   -----------------------------------------------------------------
+   Author       Leland Jansen
+   Date         November 5, 2016
+   References   T. H. Cormen, C. E. Leiserson, R. L. Rivest, and
+                  C. Stein, Introduction to Algorithms, 3rd ed.
+                  Cambridge, MA: Massachusetts Institute of
+                  Technology, 2009.
+   -----------------------------------------------------------------
+   Reliability
```

Comment:
You should write the IDs of some problems you have sucessfully solved using this implementation. Just to ensure it's not buggy.